cwms-data-api/src/test/java/cwms/cda/api/TurbineControllerIT.java

```diff
@@ -0,0 +1,313 @@
+        String json = Formats.format(Formats.parseHeader(Formats.JSON, Turbine.class), TURBINE);
+        //Create the Turbine
+        given()
+            .log().ifValidationFails(LogDetail.ALL, true)
+            .accept(Formats.JSON)
+            .contentType(Formats.JSON)
+            .body(json)
+            .header(AUTH_HEADER, user.toHeaderValue())
+            .queryParam(FAIL_IF_EXISTS, "false")
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .post("/turbines/")
+        .then()
+            .log().ifValidationFails(LogDetail.ALL, true)
+        .assertThat()
+            .statusCode(is(HttpServletResponse.SC_CREATED))
+        ;
+        String office = TURBINE.getLocation().getOfficeId();
+        // Retrieve the Turbine and assert that it exists
+        given()
+            .log().ifValidationFails(LogDetail.ALL,true)
+            .accept(Formats.JSON)
+            .queryParam(Controllers.OFFICE, office)
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .get("turbines/" + TURBINE.getLocation().getName())
+        .then()
+            .log().ifValidationFails(LogDetail.ALL,true)
+        .assertThat()
+            .statusCode(is(HttpServletResponse.SC_OK))
+            .body("location", not(nullValue()))
+            .body("project-id.name", equalTo(TURBINE.getProjectId().getName()))
+            .body("project-id.office-id", equalTo(TURBINE.getProjectId().getOfficeId()))
+        ;
+
+        // Delete a Turbine
+        given()
+            .log().ifValidationFails(LogDetail.ALL,true)
+            .accept(Formats.JSON)
+            .queryParam(Controllers.OFFICE, office)
+            .header(AUTH_HEADER, user.toHeaderValue())
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .delete("turbines/" + TURBINE.getLocation().getName())
+        .then()
+            .log().ifValidationFails(LogDetail.ALL,true)
+        .assertThat()
+            .statusCode(is(HttpServletResponse.SC_NO_CONTENT))
+        ;
+
+        // Retrieve a Turbine and assert that it does not exist
+        given()
+            .log().ifValidationFails(LogDetail.ALL,true)
+            .accept(Formats.JSON)
+            .queryParam(Controllers.OFFICE, office)
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .get("turbines/" + TURBINE.getLocation().getName())
+        .then()
+            .log().ifValidationFails(LogDetail.ALL,true)
+        .assertThat()
+            .statusCode(is(HttpServletResponse.SC_NOT_FOUND))
+        ;
+    }
+
+    @Test
+    void test_update_does_not_exist() {
+        TestAccounts.KeyUser user = TestAccounts.KeyUser.SWT_NORMAL;
+        //Create the Turbine
+        given()
+            .log().ifValidationFails(LogDetail.ALL, true)
+            .accept(Formats.JSON)
+            .contentType(Formats.JSON)
```

Comment:
contentType probably not needed here.

Reply:
cleaned up the accept vs content type in this and the embankments